TimberAPI/SpecificationSystem/Buildings/BuildingSpecificationService.cs

```diff
@@ -0,0 +1,108 @@
+        /// </summary>
+        public void Load()
+        {
+            _buildingSpecifications = _specificationService.GetSpecifications(_buildingRecipeSpecificationObjectDeserializer)
+                                                           .ToImmutableArray();
+        }
+
+        /// <summary>
+        /// Fetches all Recipes from loaded BuildingSpecifications that
+        /// have the same prefab name as given <paramref name="manufactory"/>
+        /// </summary>
+        /// <param name="manufactory"></param>
+        /// <returns></returns>
+        public IEnumerable<Recipe> GetRecipesByManufactory(Manufactory manufactory)
+        {
+            var buildingSpec = GetBuildingSpecifications(manufactory);
+            if (buildingSpec == null || buildingSpec.Recipes == null)
+            {
+                return null;
+            }
+            return buildingSpec.Recipes;
+        }
+
+        /// <summary>
+        /// Fetches all Recipes from loaded BuildingSpecifications that
+        /// have the same prefab name as given <paramref name="building"/>
+        /// </summary>
+        /// <param name="building"></param>
+        /// <returns></returns>
+        public Building GetBuildingByBuilding(Timberborn.Buildings.Building building)
+        {
+            var buildingSpec = GetBuildingSpecifications(building);
+            if (buildingSpec == null || buildingSpec.Building == null)
+            {
+                return null;
+            }
+            return buildingSpec.Building;
+        }
+
+        /// <summary>
+        /// Fetches all Recipes from loaded BuildingSpecifications that
+        /// have the same prefab name as given <paramref name="mechanicalNodeSpecification"/>
+        /// </summary>
+        /// <param name="mechanicalNodeSpecification"></param>
+        /// <returns></returns>
+        public MechanicalNode GetMechanicalNodeByMechanicalNodeSpecification(
+            MechanicalNodeSpecification mechanicalNodeSpecification)
+        {
+            var buildingSpec = GetBuildingSpecifications(mechanicalNodeSpecification);
+            if (buildingSpec == null || buildingSpec.MechanicalNode == null)
+            {
+                return null;
+            }
+
+            return buildingSpec.MechanicalNode;
+        }
+
+        /// <summary>
+        /// Helper function returns the specification that has the same
+        /// prefab name as given <paramref name="gameObject"/>
+        /// </summary>
+        /// <param name="gameObject"></param>
+        /// <returns></returns>
+        private BuildingSpecification GetBuildingSpecifications(MonoBehaviour gameObject)
+        {
+            var prefab = gameObject.GetComponent<Prefab>();
+            var prefabName = prefab.PrefabName;
+            if (_buildingSpecifications == null)
+            {
+                return null;
+            }
+            var specs = _buildingSpecifications.Where(x => x?.BuildingId == prefabName);
```

Comment:
Personally find it nicer if such short LinQ with no extra logic is returned immidiately.
`return _buildingSpecifications.Where(x => x?.BuildingId == prefabName).FirstOrDefault();`

Reply:
Tbh same. In this case it's separate because I did some debugging and forgot to change it afterwards